tests/osaka/eip7939_count_leading_zeros/test_count_leading_zeros.py

```diff
@@ -222,3 +222,62 @@ def test_clz_fork_transition(blockchain_test: BlockchainTestFiller, pre: Alloc):
             ),
         },
     )
+
+
+class CallingContext:
+    """Context for calling operations."""
+
+    callee_context = 1  # CALL
+    caller_context = 2  # DELEGATECALL
+    no_context = 3  # STATICCALL
+
+
+@pytest.mark.valid_from("Osaka")
+@pytest.mark.parametrize("bits", [0, 16, 64, 128, 255])
+@pytest.mark.parametrize(
+    "opcode,context",
+    [
+        pytest.param(Op.CALL, CallingContext.callee_context, id="call"),
+        pytest.param(Op.DELEGATECALL, CallingContext.caller_context, id="delegatecall"),
+        pytest.param(Op.CALLCODE, CallingContext.caller_context, id="callcode"),
+        pytest.param(Op.STATICCALL, CallingContext.no_context, id="staticcall"),
+    ],
+)
+def test_clz_call_operation(
+    state_test: StateTestFiller, pre: Alloc, opcode: Op, context: CallingContext, bits: int
+):
+    """Test CLZ opcode with call operation."""
+    callee_code = Op.PUSH32(1 << bits) + Op.DUP1
+
+    if context != CallingContext.no_context:
+        callee_code += Op.CLZ + Op.PUSH0 + Op.SSTORE
+
+    callee_code += Op.CLZ + Op.PUSH0 + Op.MSTORE + Op.RETURN(0, 32)
```

Comment:
```suggestion
    callee_code = Op.PUSH32(1 << bits) + Op.CLZ

    if context != CallingContext.no_context:
        callee_code += Op.DUP1 + Op.PUSH0 + Op.SSTORE

    callee_code += Op.PUSH0 + Op.MSTORE + Op.RETURN(0, 32)
```
I feel like this is a tiny bit more readable (do the CLZ operation then duplicate the result, instead of redoing the operation).